lib/assets/javascripts/markdown-it-diaspora-linkify.js

```diff
@@ -2,69 +2,11 @@
-},{}],4:[function(require,module,exports){
 // Replace link-like texts with link nodes.
 //
 'use strict';
 
-var urlRegex = require('url-regex');
-var LINK_SCAN_RE = /www|@|\:\/\//;
+var urlRegex = /(?:^|\s)(?:(?:https?|ftp):\/\/|mailto:|xmpp:|www[^.\s]*\.)\S+/gi;
```

Comment:
I fear this one is a bit too open though: https://www.debuggex.com/r/8859oQ_AME577XSr

We're currently using [twitter-text-rb](https://github.com/twitter/twitter-text) in the backend for URL recognition, it also has a [JS implementation](https://github.com/twitter/twitter-text/tree/master/js), I think it might be worth investigating what they use or maybe even utilizing the library.